libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/SyncUpTarget.java

```diff
@@ -245,24 +255,64 @@ public int updateOnServer(SyncManager syncManager, String objectType, String obj
+        final String objectType = (String) SmartStore.project(record, Constants.SOBJECT_TYPE);
+        final String objectId = record.getString(getIdFieldName());
 
-            final String query = SOQLBuilder.getInstanceWithFields(getModificationDateFieldName())
-                    .from(objectType)
-                    .where(getIdFieldName() + " = '" + objectId + "'")
-                    .build();
+        RestRequest lastModRequest = RestRequest.getRequestForRetrieve(syncManager.apiVersion, objectType, objectId, Arrays.asList(new String[]{getModificationDateFieldName()}));
+        RestResponse lastModResponse = syncManager.sendSyncWithSmartSyncUserAgent(lastModRequest);
 
-            RestResponse lastModResponse = syncManager.sendSyncWithSmartSyncUserAgent(RestRequest.getRequestForQuery(syncManager.apiVersion, query));
-            JSONArray records = lastModResponse.asJSONObject().optJSONArray(Constants.RECORDS);
-            return records != null && records.length() > 0 ? records.optJSONObject(0).optString(getModificationDateFieldName()) : null;
+        return new RecordModDate(
+                lastModResponse.isSuccess() ? lastModResponse.asJSONObject().getString(getModificationDateFieldName()) : null,
+                lastModResponse.getStatusCode() == HttpURLConnection.HTTP_NOT_FOUND
+        );
+    }
+
+    /**
+     * Return true record is more recent than corresponding record on server
+     * NB: also return true if both were deleted or if local mod date is missing
+     *
+     * Used to decide whether a record should be synced up or not when using merge mode leave-if-changed
+     *
+     * @param syncManager
+     * @param record
+     * @return
+     * @throws JSONException
+     * @throws IOException
+     */
+    public boolean isNewerThanServer(SyncManager syncManager, JSONObject record) throws JSONException, IOException {
+        if (isLocallyCreated(record)) {
+            return true;
         }
-        catch (Exception e) {
-            // Caller expects null to be returned if the last modified date could not be fetched
-            return null;
+
+        final RecordModDate localModDate = new RecordModDate(
+                JSONObjectHelper.optString(record, getModificationDateFieldName()),
+                isLocallyDeleted(record)
+        );
+
+        final RecordModDate remoteModDate = fetchLastModifiedDate(syncManager, record);
+
+        return isNewerThanServer(localModDate, remoteModDate);
+    }
+
+    /**
+     * Return true if local mod date is greater than remote mod date
+     * NB: also return true if both were deleted or if local mod date is missing
+     *
+     * @param localModDate
+     * @param remoteModDate
+     * @return
+     */
+    protected boolean isNewerThanServer(RecordModDate localModDate, RecordModDate remoteModDate) {
+
+        if ((localModDate.timestamp != null && remoteModDate.timestamp != null
```

Comment:
Here is where we are smarter than before deciding whether the record should be skipped during leave-if-changed.